soco/core.py

```diff
@@ -1833,6 +1873,185 @@ def add_item_to_sonos_playlist(self, queueable_item, sonos_playlist):
             ('AddAtIndex', 4294967295)
         ])
 
+    def reorder_sonos_playlist(self, object_id, tracks, new_pos, update_id=0):
```

Comment:
Not only here, but in all methods. I like that you have made the argument that specifies the Sonos playlist flexible, so that it takes both an ID and the playlist object, but please turn the logic around, so that the playlist object is the first class citizen and the ID the alternative, both in the argument name and in the phrasing in the docstring. As I mentioned in one of the other PRs, I think one of the goals of the SoCo API should be to elegantly mask some of all the gory details of Sonos by clear Pythonic types and concepts and I really feel that this falls in that category.

I will make smaller comments with the keyword playlist_object_first in the places were I think there should be a modification.